lib/Target/HSACO/HSACOTranslation.cpp

```diff
@@ -31,8 +31,10 @@
 #include "llvm/Transforms/Scalar.h"
 #include "llvm/Transforms/Utils/Cloning.h"
 #include <iostream>
+#include <dlfcn.h>
```

Comment:
I used dladdr to find the correct amdgpu.ids file.  This is a more efficient way, as the old method of looking at `/proc/self/exe` was putting us in `/usr/bin/python` and then walking back out.  This works if we are making an assumption that we will always be in `/usr`  If we are testing locally etc. It will cause headaches as the python binary stays where it is, but our `so's` may be in our repository or other location.  We now search at the location of the `libdrm` shared object and deeper.  

This also saves us from having to walk the entire `/usr` filesystem trying to find amdgpu.  Not sure if this would have applications on the pytorch side as well? not sure we've measured if that lookup takes any time at all?